packages/webdriverio/src/utils/index.ts

```diff
@@ -427,6 +432,14 @@ export async function findDeepElements(
     return deepElementResult as ElementReference[]
 }
 
+/**
+* Temporary patch for https://github.com/mozilla/geckodriver/issues/2223
+*/
+function returnUniqueNodes(nodes: ExtendedElementReference[]): ExtendedElementReference[] {
+    const ids = new Set()
+    return nodes.filter((node) => !ids.has(node[ELEMENT_KEY]) && ids.add(node[ELEMENT_KEY]))
```

Comment:
I actually just saw a small thing at the end, we now filter but don't do anything with it, instead of assigning to the Set, you can filter and what is returned can be put into the Set instead 👍

```suggestion
    return new Set(nodes.filter((node) => !ids.has(node[ELEMENT_KEY]) && ids.add(node[ELEMENT_KEY])))
```